proxy/create_container_interceptor.go

```diff
@@ -94,34 +99,33 @@ func (i *createContainerInterceptor) setWeaveDNS(container *createContainerReque
 	name := r.URL.Query().Get("name")
 	if container.Hostname == "" && name != "" {
 		container.Hostname = name
-		container.Domainname = i.getDNSDomain()
+		container.Domainname = dnsDomain
 	}
 
 	return nil
 }
 
-func (i *createContainerInterceptor) getDNSDomain() (domain string) {
-	domain = nameserver.DefaultLocalDomain
+func (i *createContainerInterceptor) getDNSDomain() (string, bool) {
 	dnsContainer, err := i.proxy.client.InspectContainer("weavedns")
 	if err != nil ||
 		dnsContainer.NetworkSettings == nil ||
 		dnsContainer.NetworkSettings.IPAddress == "" {
-		return
+		return nameserver.DefaultLocalDomain, false
 	}
 
 	url := fmt.Sprintf("http://%s:%d/domain", dnsContainer.NetworkSettings.IPAddress, nameserver.DefaultHTTPPort)
 	resp, err := http.Get(url)
 	if err != nil || resp.StatusCode != http.StatusOK {
-		return
+		return nameserver.DefaultLocalDomain, false
 	}
 	defer resp.Body.Close()
 
 	b, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return
+		return nameserver.DefaultLocalDomain, false
 	}
 
-	return string(b)
+	return string(b), true
```

Comment:
one of the few places where named returns would work well, i.e. all the `false` cases can just be `return`.